library/repo.go

```diff
@@ -621,6 +637,14 @@ func (r *Repo) SetPipelineType(v string) {
 	r.PipelineType = &v
 }
 
+func (r *Repo) SetPreviousName(v string) {
+	// return if Repo type is nil
+	if r == nil {
+		return
+	}
+	r.PreviousName = &v
```

Comment:
I think we're missing a space here:

```suggestion
	}

	r.PreviousName = &v
```